packages/scrollable-image/ui/package.json

```diff
@@ -0,0 +1,23 @@
+{
+  "name": "scrollableImage-ui",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "scripts": {
+    "build": "make build",
+    "dev": "make dev"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.9.7",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.47",
+    "express": "^4.17.1",
+    "path": "^0.12.7",
+    "prop-types": "^15.7.2",
+    "react": "^16.3.0"
+  },
+  "devDependencies": {
+    "html-webpack-plugin": "^4.0.1",
+    "react-dom": "^16.3.0"
+  }
+}
```

Comment:
Do we support nested monorepo?
Would this sub-package break the lerna conventions?

Reply:
It seems that lerna doesn't support nested packages in its convention [[ref]](https://github.com/lerna/lerna/issues/1736).

At first I plan to let `ui` sub-package to be independent of lerna package management. Since it is out of sync to lerna packages, I believe there wouldn't  exist circular dependencies issue. 

However, things goes different when `scrollable-image-ui` imports `@twreporter/orangutan` instead of using `src` in `@twreporter/scrollable-image`. Manually upgrade dependencies would be needed without lerna's managing. 

It will be updated to a sub-package under `orangutan-monorepo/packages/ `.

Reply:
Hold on a second.
Is it ok to expose `ui` react components in `@twreporter/scrollable-image`?

Reply:
The sub-package `ui` will be moved to `orangutan-monorepo/packages/scrollable-image-ui` and published as `@twreporter/scrollable-image-ui`. 

Folder `public/` contains files built by webpack, which will be the published folder. 

Do you have a concern about exposing `ui` react components in `@twreporter/scrollable-image-ui` ?

Reply:
After thinking twice,
building `@twreporter/scrollable-image-ui` is simpler and more easy-to-understand than exposing `ui` component in `@twreporter/scrollable-image`.


### Remarks
The following is showing how I think about this issue.
If you are interested, we can have further discussion.

We can expose `ui` react components in `@twreporter/scrollable-image`.
For example:
```
// orangutan-monorepo/packages/scrollable-image/src/index.js
// mv `ui/src/components` to `src/ui/components`
import BuildEmbeddedCodeComponent from './ui/components/'

export default {
  BuildEmbeddedCodeComponent,
  Component,
  buildEmbeddedCode,
  getWebpackEntry: () => {
    return path.resolve(__dirname, './build-code/client.js'),
  },
  getTestData: () => {
    return {
      data: mockImgSrcs,
      lazyload: true,
    }
  },
}
```

In `scrollabel-image/dev` folder, we can have multiple entry points to build scrollable-image page and build-scrollable-image-embedded-code page. Thus, we still can use `react-hot-loader`, `webpack-dev-server` to develop.

In cloud functions,
in `package.json`
```json
"scripts": {
  "gcp-build": "make build-webpack",
}
```
in `Makefile`
```
build-webpack:
  webpack --config webpack.config.js
```

in `webpack.config.js`,
we config cloud functions entry page, output, module ...etc.
And we can use `HtmlWebpackPlugin` to build `index.html` with webpack bundles.

in `entry-point.js`
```javascript
const orangutan = require('@twreporter/orangutan').default
const scrollableImage = orangutan.scrollableImage
const React == require('react')
const ReactDOM = require('react-dom')
const actions = require('./api/actions')

ReactDOM.render(
<scrollableImage.BuildEmbeddedCodeComponent 
  submit={actions.collectImages} // send user inputs back to api 
/>, document.getElementById('root'))
```

we can 
```javascript 

exports buildScrollableEmbeddedCode = (req, res) => {
  case '/': {
    const scrollableImage = orangutan.scrollableImage
    res.sendFile('index.html', options, callback)
  }
  case '/api': {
    // handle user inputs
    // return embedded code.
  }
}
```

Reply:
Nice elaboration indeed !!

After considering following reasons, I prefer building `@twreporter/scrollable-image-ui` as well:
1. keep the webpack config in `dev` folder have one objective might be helpful to maintenance. (e.g. test-embedded-code)
2. `BuildEmbeddedCodeComponent` seems not like a part of `scrollable-image`. In my opinion, it uses `@twreporter/scrollable-image` to build an application named `ui` but not belongs to it. Therefore I am not a fan of exporting it in `scrollable-image`.
3. Use `@twreporter/scrollable-image-ui` option, we can keep the interface between `orangutan` and other packages (`orangutan-monorepo/packages/scrollable-image/src/index.js`) simple and consistent.

Reply:
👍 👍 

We could close this PR,
and reopen it when it is ready.